### ISSUE TITLE
Unable to use boost/log/utility/setup

### ISSUE THREAD

**Issue by ctrysbita** [nelhage/rules_boost#151]
When using boost log utility to setup a console log with format, linker reported an error:
```bash
function boost::shared_ptr<boost::log::v2s_mt_posix::sinks::synchronous_sink<boost::log::v2s_mt_posix::sinks::basic_text_ostream_backend<char> > > boost::log::v2s_mt_posix::aux::add_console_log<char, boost::parameter::aux::tagged_argument_list_of_1<boost::parameter::aux::tagged_argument<boost::log::v2s_mt_posix::keywords::tag::format, char const [1]> > >(std::basic_ostream<char, std::char_traits<char> >&, boost::parameter::aux::tagged_argument_list_of_1<boost::parameter::aux::tagged_argument<boost::log::v2s_mt_posix::keywords::tag::format, char const [1]> > const&): 
error: undefined reference to 'boost::log::v2s_mt_posix::basic_formatter<char> boost::log::v2s_mt_posix::parse_formatter<char>(char const*, char const*)'
```

Code to setup:
```cpp
#include <boost/log/utility/setup/console.hpp>
boost::log::add_console_log(std::cout, boost::log::keywords::format = "");
```

**Comment by reltuk** [comment]
This looks to be because we compile with (and export defines of) `BOOST_LOG_WITHOUT_DEFAULT_FACTORIES` and `BOOST_LOG_WITHOUT_SETTINGS_PARSERS`, so the setting parsers aren't compiled.

I've opened https://github.com/nelhage/rules_boost/pull/152, which fixes it locally. That PR may initially fail against the windows CI, but hopefully I can fix it soon.

**Comment by nelhage** [comment]
Resolved by #152
